packages/manager/apps/ips/src/utils/getCatalog.ts

```diff
@@ -0,0 +1,136 @@
+import { useQuery } from '@tanstack/react-query';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { useContext } from 'react';
+
+export const BYOIP_FAILOVER_V4 = 'byoip-failover-v4';
+export const CONFIG_NAME = {
+  CAMPUS: 'campus',
+  IPRIR: 'ipRir',
+} as const;
+
+export type Campus = {
+  name: string;
+  planCode: string;
+};
+
+export type ProductConfiguration = {
+  name: string;
+  values: string[] | Campus[];
+};
+
+export type Product = {
+  description: string;
+  internalType: string;
+  name: string;
+  configurations: ProductConfiguration[];
+};
+
+export type PlanDetails = {
+  metadatas: Record<string, unknown>;
+  pricings: Record<string, unknown>;
+  product: Product;
+};
+
+export type Plan = {
+  addonsFamily: string;
+  consumptionBillingStrategy: string;
+  details: PlanDetails;
+  invoiceName: string;
+  pricingType: string;
+  planCode: string;
+};
+
+export type CatalogResponse = {
+  plans: Plan[];
+};
+
+export const useCatalog = () => {
+  const { shell } = useContext(ShellContext);
+  const { environment } = shell;
+
+  return useQuery({
+    queryKey: ['catalog', 'bringYourOwnIp'],
+    queryFn: async () => {
+      const env = await environment.getEnvironment();
+      const { ovhSubsidiary } = env.getUser();
+
+      const { data } = await apiClient.v6.get<CatalogResponse>(
+        '/order/catalog/formatted/bringYourOwnIp',
+        {
+          params: {
+            ovhSubsidiary,
+          },
+        },
+      );
+
```

Comment:
You should have 2 separate hooks :
- `useByoipCatalog(subsidiary) which does the query. Take the subsidiary as a parameter to avoid bugs in case someday we succeed to change the subsidiary without reloading the app. It should also be in the queryKey and you can add `enabled: !!ovhSubsidiary` to avoid doing the call without a subsidiary.
- `useConfiguration(configurationName)` that uses the `useByoipCatalog` hook and returns the right ProductConfiguration